packages/common/src/services/integrations/jwp/base/JWPAPIService.ts

```diff
@@ -0,0 +1,128 @@
+import { inject, injectable } from 'inversify';
+
+import StorageService from '../../../StorageService';
+
+const INPLAYER_TOKEN_KEY = 'inplayer_token';
+const INPLAYER_IOT_KEY = 'inplayer_iot';
+
+const CONTENT_TYPES = {
+  json: 'application/json',
+  form: 'application/x-www-form-urlencoded',
+};
+
+type RequestOptions = {
+  withAuthentication?: boolean;
+  keepalive?: boolean;
+  contentType?: keyof typeof CONTENT_TYPES;
+  responseType?: 'json' | 'blob';
+  includeFullResponse?: boolean;
+};
+
+@injectable()
+export default class JWPAPIService {
+  private readonly storageService: StorageService;
+
+  private sandbox = true;
+
+  siteId = '';
+
+  constructor(@inject(StorageService) storageService: StorageService) {
+    this.storageService = storageService;
+  }
+
+  setToken = (token: string, refreshToken = '', expires: number) => {
+    return this.storageService.setItem(INPLAYER_TOKEN_KEY, JSON.stringify({ token, refreshToken, expires }), false);
+  };
+
+  getToken = async () => {
+    const tokenObject = await this.storageService.getItem(INPLAYER_TOKEN_KEY, true, false);
+
+    if (tokenObject) {
+      return tokenObject as { token: string; refreshToken: string; expires: number };
+    }
+
+    return { token: '', refreshToken: '', expires: 0 };
+  };
+
+  removeToken = async () => {
+    await Promise.all([this.storageService.removeItem(INPLAYER_TOKEN_KEY), this.storageService.removeItem(INPLAYER_IOT_KEY)]);
+  };
+
+  isAuthenticated = async () => {
+    const token = await this.getToken();
+
+    return !!token.token && token.expires > Date.now() / 1000;
+  };
+
+  private getBaseUrl = () => (this.sandbox ? 'https://staging-v2.inplayer.com' : 'https://services.inplayer.com');
+
+  private performRequest = async (
+    path: string = '/',
+    method = 'GET',
+    body?: Record<string, unknown>,
+    { contentType = 'form', responseType = 'json', withAuthentication = false, keepalive, includeFullResponse = false }: RequestOptions = {},
+  ) => {
+    const headers: Record<string, string> = {
+      'Content-Type': CONTENT_TYPES[contentType],
+    };
+
+    if (withAuthentication) {
+      const tokenObject = await this.getToken();
+
+      if (tokenObject.token) {
+        headers.Authorization = `Bearer ${tokenObject.token}`;
+      }
+    }
+
+    const _body =
+      body &&
+      Object.entries(body)
+        .map(([key, value]) =>
+          key && (value || value === 0)
+            ? value.toString() === '[object Object]'
+              ? Object.entries(value)
+                  .map(([subkey, subval]) => `${key}[${subkey}]=${encodeURIComponent(subval)}`)
+                  .join('&')
+              : `${key}=${typeof value === 'string' ? encodeURIComponent(value) : value}`
+            : '',
+        )
+        .filter(Boolean)
+        .join('&');
+
+    const endpoint = /^\//.test(path) ? `${this.getBaseUrl()}${path}` : path;
```

Comment:
This is more semantical (easier to understand what is happening).

```suggestion
    const endpoint = path.startsWith('/') ? `${this.getBaseUrl()}${path}` : path;
```

Although, this is more safe (i.e. when passing a path without a starting slash)

```suggestion
    const endpoint = path.startsWith('http') ? path : `${this.getBaseUrl()}${path}`;
```